e2e/framework.go

```diff
@@ -193,6 +204,58 @@ func (c *cluster) WaitForHeight(num uint64, timeout time.Duration, nodes ...[]st
 	}
 }
 
+func (c *cluster) getNodeHeight(n *node) uint64 {
```

Comment:
Now that I think about this, do we really need to get the proposal to get the Height? Shouldn't the height be always equal to the index? I think we can remove a noisy lock if we do it that way.

Reply:
Height is not equal to index because, initial height is always 1 since 0 us genesis, 
First inserted height would be 2 on index 0. We could do some kind of calculation because difference between height and index is 2.